src/dsl.rs

```diff
@@ -33,6 +33,10 @@ impl<F, TraceArgs, StepArgs> CircuitContext<F, TraceArgs, StepArgs> {
         Queriable::Forward(self.sc.add_forward(name, phase), false)
     }
 
+    pub fn shared(&mut self, name: &str) -> Queriable<F> {
```

Comment:
Please add shared_with_phase to the DSL API

Reply:
I will merge this for now and fix this in Part 2, so that I only need to sync this change to Part 3.